.github/workflows/PR-build.yml

```diff
@@ -0,0 +1,141 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: MIT
+
+name: PR Build
+on:
+  pull_request:
+    branches: 
+      - main 
+    types: 
+      - synchronize
+      - reopened
+      - ready_for_review
+    paths-ignore:
+      - '**/*.md'
+      - '.github/**'
+      - '!.github/workflows/PR-build.yml'
```

Comment:
I am not going to approve a PR that turns off building and unit testing on the main branch after a merge. If other people want to, that's fine.

Reply:
Question from me: would it be better to change the `integrationTest.yml` to `CI.yml` and add the unit testing on the `CI.yml` ? @SaxyPandaBear

Reply:
This hasn't changed in 2 months, and my stance on it hasn't either. If other people want to approve this, then I will too but I don't agree with this change.

Reply:
Yes I changed that during the past 2 months. We always run `integrationTest` when merge from main, therefore, I [added a unit test for Linux](https://github.com/aws/amazon-cloudwatch-agent/pull/665/files#diff-3ab46ee209a127470fce3c2cf106b1a1dbadbb929a4b5b13656a4bc4ce19c0b8R89-R91)  and for [Mac](https://github.com/aws/amazon-cloudwatch-agent/pull/665/files#diff-3ab46ee209a127470fce3c2cf106b1a1dbadbb929a4b5b13656a4bc4ce19c0b8R288-R291)

Reply:
If you disagree with running make test  in `integrationTest.yml`, then I'm fine with adding it back too. FYI,  I have added back the push on main branch